yarn-project/archiver/src/archiver/data_retrieval.ts

```diff
@@ -122,7 +180,11 @@ export async function processL2BlockProposedLogs(
         timestamp: await getL1BlockTime(publicClient, log.blockNumber),
       };
 
-      retrievedBlocks.push({ ...block, l1 });
+      const chainId = new Fr(await publicClient.getChainId());
+
+      const version = new Fr(await rollup.read.getVersion({ blockNumber: log.blockNumber }));
```

Comment:
When this is passing along `blockNumber` as this, it will likely try to run at that point in time, which could make it explode as historical execution.